quinn-proto/src/packet.rs

```diff
@@ -612,6 +610,15 @@ impl PlainHeader {
     }
 }
 
+#[derive(Clone, Debug)]
+pub(crate) struct PlainHeaderInitial {
```

Comment:
Naming nit: maybe `InitialHeader`? I don't think `Plain` provides sufficient disambiguation value here, and would be nice to have a more natural word order.

Reply:
This is the pre-decryption view of the header; see also `enum PlainHeader` vs. `enum Header`. I'll go with `PlainInitialHeader`.